text/shaper_test.go

```diff
@@ -35,19 +35,19 @@ func TestWrappingTruncation(t *testing.T) {
 			MaxLines:  i,
 		}, 200, 200, english, textInput)
 		lineCount := len(cache.txt.lines)
-		glyphs := []Glyph{}
-		for g, ok := cache.NextGlyph(); ok; g, ok = cache.NextGlyph() {
-			glyphs = append(glyphs, g)
-		}
-		if i <= untruncatedCount {
-			if lineCount != i {
-				t.Errorf("expected %d lines, got %d", i, lineCount)
-			}
-		} else if i > untruncatedCount {
-			if lineCount != untruncatedCount {
-				t.Errorf("expected %d lines, got %d", untruncatedCount, lineCount)
+		for {
+			_, ok := cache.NextGlyph()
+			if !ok {
+				break
 			}
 		}
+		min := i
+		if min > untruncatedCount {
+			min = untruncatedCount
+		}
+		if lineCount != min {
+			t.Errorf("expected %d lines, got %d", min, lineCount)
+		}
```

Comment:
@whereswaldon can you take a look at this change?

Reply:
This change is valid, but mostly points out that I've checked whether truncation occurred using an unexported implementation detail instead of the exported API. I think I'll rework this test case to count the lines using the glyph stream instead.

Reply:
I've sent [this patch](https://lists.sr.ht/~eliasnaur/gio-patches/patches/39302) to cover simplifying the test. Thanks @jfcg for pointing out that this can be improved!

I don't especially feel the need to rewrite all of the glyph iteration loops to use the style proposed here. It is simpler on some levels, but many more lines. When/if the custom iterators proposal is accepted, the multiple calls to NextGlyph will collapse into one invocation anyway.